src/Material/Menu.elm

```diff
@@ -0,0 +1,523 @@
+                ++ "px"
+
+      , flip (css' "left") (((model.alignment == topLeft) || (model.alignment == bottomLeft)) && isJust model.oracle) <|
+
+          case model.oracle of
+            Nothing -> "auto"
+            Just oracle ->
+
+              toString oracle.menu.offsetLeft ++ "px"
+      ]
+      [ styled div
+        [ cs "mdl-menu__outline"
+
+        , css' "width"
+            ( case model.oracle of
+                Just oracle ->
+                  oracle.menu.bounds.width
+                  |> toString
+                  |> flip (++) "px"
+                Nothing     -> "auto"
+            )
+            ( isJust model.oracle )
+
+        , css' "height"
+            ( case model.oracle of
+                Just oracle ->
+                  oracle.menu.bounds.height
+                  |> toString
+                  |> flip (++) "px"
+                Nothing     -> "auto"
+            )
+            ( isJust model.oracle )
+
+        , model.alignment
+        ]
+        [
+        ]
+      , styled ul
+        [ cs "mdl-menu"
+        , cs "mdl-js-menu"
+        , model.alignment
+        , cs' "is-animating" (model.open == Just False)
+        , css' "clip"
+          (case model.oracle of
+
+             Nothing     -> "auto"
+             Just oracle ->
+
+               let
+                 rect x y w h =
+                   String.join ""
+                   [ "rect( "
+                   , toString x ++ "px"
+                   , " "
+                   , toString y ++ "px"
+                   , " "
+                   , toString w ++ "px"
+                   , " "
+                   , toString h ++ "px"
+                   , ")"
+                   ]
+
+                 width  = oracle.menu.bounds.width
+                 height = oracle.menu.bounds.height
+               in
+                 if model.open == Just True then
+
+                     rect 0 width height 0
+
+                   else
+
+                     if model.alignment == bottomRight then
+                         rect 0 width 0 width
+                       else if model.alignment == topLeft then
+                         rect height 0 height 0
+                       else if model.alignment == topRight then
+                         rect height width height width
+                       else
+                         ""
+          )
+          (isJust model.oracle)
+        ]
+        ( let
+
+            makeItem n item =
+              let
+                transitionDuration =
+                  constant.transitionDurationSeconds *
+                  constant.transitionDurationFraction
+
+                offsetTop n =
+                  case model.oracle of
+                    Nothing -> 0
+                    Just oracle ->
+                      oracle.offsetTops
+                      |> List.drop (n-1) -- n is 1-based
+                      |> List.head
+                      |> fromJust "Menu.view: offsetTop"
```

Comment:
I see what you are doing, but I really don't think `fromJust` is something we want. The promise of Elm is no crashes; if we happen to be wrong and the oracle __is__ missing an element, I'd rather have weird run-time behaviour than a crash.